test/functional/test_framework/test_node.py

```diff
@@ -98,14 +98,23 @@ def __init__(self, i, datadir, *, chain, rpchost, timewait, bitcoind, bitcoin_cl
             "-uacomment=testnode%d" % i,
         ]
         if use_valgrind:
-            default_suppressions_file = os.path.join(
-                os.path.dirname(os.path.realpath(__file__)),
-                "..", "..", "..", "contrib", "valgrind.supp")
-            suppressions_file = os.getenv("VALGRIND_SUPPRESSIONS_FILE",
-                                          default_suppressions_file)
-            self.args = ["valgrind", "--suppressions={}".format(suppressions_file),
-                         "--gen-suppressions=all", "--exit-on-first-error=yes",
-                         "--error-exitcode=1", "--quiet"] + self.args
+            valgrind_version = self.get_valgrind_version()
+            exit_on_first_error_flag = ""
+            if valgrind_version and valgrind_version >= MIN_VALGRIND_VERSION_EARLY_EXIT:
```

Comment:
I think the logic is wrong here. This will only run `valgrind` if `valgrind_version >= MIN_VALGRIND_VERSION_EARLY_EXIT` :)

The only impact `valgrind_version >= MIN_VALGRIND_VERSION_EARLY_EXIT` should have is to enable `--exit-on-first-error=yes`. `self.args` should be set both for old and new versions of Valgrind :)

Reply:
Yeah, just saw it. Will reorganize the code.